vault/data_source_gcp_auth_backend_role.go

```diff
@@ -122,9 +122,9 @@ func gcpAuthBackendRoleRead(d *schema.ResourceData, meta interface{}) error {
 	log.Printf("[DEBUG] Read gcp auth backend role %q ID", path)
 
 	if resp == nil {
-		d.SetId("")
```

Comment:
I see that there are other places where we explicitly call `d.SetId("")` before returning if the value (e.g. `resp`) that we're about to reference is nil. How come we don't have to in this case?

Reply:
Good catch!

I did a bit of digging and this should actually be an error, since the role does exist.

PR on its way!